File: Example/PrebidDemoJava/src/main/java/org/silvermob/mobile/javademo/activities/BaseAdActivity.java
package org.silvermob.mobile.javademo.activities;

import android.os.Bundle;
import android.view.ViewGroup;

import androidx.annotation.CallSuper;
import androidx.annotation.Nullable;
import androidx.appcompat.app.AppCompatActivity;
import androidx.databinding.DataBindingUtil;

import org.silvermob.mobile.javademo.R;
import org.silvermob.mobile.javademo.databinding.ActivityDemoBinding;
import org.silvermob.mobile.javademo.testcases.TestCase;
import org.silvermob.mobile.javademo.testcases.TestCaseRepository;
import org.silvermob.mobile.javademo.utils.Settings;

public class BaseAdActivity extends AppCompatActivity {

    protected ActivityDemoBinding binding;

    private TestCase testCase = TestCaseRepository.lastTestCase;

    @Override
    @CallSuper
    public void onCreate(@Nullable Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        binding = DataBindingUtil.setContentView(this, R.layout.activity_demo);
        binding.tvTestCaseName.setText(getString(testCase.getTitleStringRes()));
    }

    protected ViewGroup getAdWrapperView() {
        return binding.frameAdWrapper;
    }

    protected int getRefreshTimeSeconds() {
        return Settings.get().getRefreshTimeSeconds();
    }

}

File: Example/PrebidDemoKotlin/src/main/java/org/silvermob/mobile/prebidkotlindemo/activities/BaseAdActivity.kt
package org.silvermob.mobile.prebidkotlindemo.activities

import android.os.Bundle
import android.view.ViewGroup
import androidx.annotation.CallSuper
import androidx.appcompat.app.AppCompatActivity
import androidx.databinding.DataBindingUtil
import org.silvermob.mobile.prebidkotlindemo.R
import org.silvermob.mobile.prebidkotlindemo.databinding.ActivityDemoBinding
import org.silvermob.mobile.prebidkotlindemo.testcases.TestCase
import org.silvermob.mobile.prebidkotlindemo.testcases.TestCaseRepository
import org.silvermob.mobile.prebidkotlindemo.utils.Settings

open class BaseAdActivity : AppCompatActivity() {

    /**
     * ViewGroup container for any ad view.
     */
    protected val adWrapperView: ViewGroup
        get() = binding.frameAdWrapper

    /**
     * Seconds for auto-refreshing any banner ad.
     */
    protected val refreshTimeSeconds: Int
        get() = Settings.get().refreshTimeSeconds

    private lateinit var binding: ActivityDemoBinding
    private var testCase: TestCase = TestCaseRepository.lastTestCase

    @CallSuper
    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        binding = DataBindingUtil.setContentView(this, R.layout.activity_demo)
        binding.tvTestCaseName.text = getText(testCase.titleStringRes)
    }

}